documents/oauth_token.md

```diff
@@ -0,0 +1,140 @@
+### OAuthToken
+
+```rb
+require "razorpay"
+```
+
+### Generate Authorize Url
+```rb
+body = {
+    submerchant_id: '<SUBMERCHANT_MID>',
+    timestamp: Time.now.to_i
+}
+onboarding_signature = Razorpay::Utility.generate_onboarding_signature(body, '<YOUR_CLIENT_SECRET>')
+
+options = {
+    'client_id'            => '<YOUR_CLIENT_ID>',
+    'redirect_uri'         => 'https://example.com/razorpay_callback',
+    'scopes'               => ["read_write"],
+    'state'                => 'NOBYtv8r6c75ex6WZ',
+    'onboarding_signature' => onboarding_signature
+}
+authorize_url = Razorpay::OAuthToken.get_auth_url(options)
+```
+
+**Parameters:**
+| Name                 | Type   | Description                                                                                                                                             |
+|----------------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| client_id*           | string | Unique client identifier.                                                                                                                               |
+| redirect_uri*        | string | Callback URL used by Razorpay to redirect after the user approves or denies the authorisation request. The client should whitelist the 'redirect_uri'.  |
+| scopes*              | array  | Defines what access your application is requesting from the user. You can request one or multiple scopes by adding them to an array as indicated above. |
+| state*               | string | A random string generated by your service. This parameter helps prevent cross-site request forgery (CSRF) attacks.                                      |
+| onboarding_signature | string | A cryptographic string generated by your service using generateOnboardingSignature method in Utils class. Only applicable for accounts created with pre-fill KYC |
+
+**Response:**
+```
+"https://auth.razorpay.com/authorize?response_type=code&client_id=<YOUR_CLIENT_ID>&redirect_uri=https:%2F%2Fexample.com%2Frazorpay_callback&scope[]=read_only&scope[]=rx_read_write&state=NOBYtv8r6c75ex6WZ&onboarding_signature=<GENERATED_ONBOARDING_SIGNATURE>"
+```
+
+-------------------------------------------------------------------------------------------------------
+### Get Access token
+```rb
+options = {
+    'client_id'          => '<YOUR_CLIENT_ID>',
+    'client_secret'     => '<YOUR_CLIENT_SECRET>',
+    'redirect_uri'      => 'https://example.com/razorpay_callback',
+    'grant_type'        => 'authorization_code',
+    'code'              => '<AUTHORIZATION_CODE>',
+    'mode'              => 'test'
+}
+oauth_token = Razorpay::OAuthToken.get_access_token(options)
+```
+
+**Parameters:**
```

Comment:
Same with this format.

Reply:
Formatting was done. Had to add additional line after parameters which was causing this issue.